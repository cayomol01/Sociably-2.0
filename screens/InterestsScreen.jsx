import React, {useState} from 'react';
import {StyleSheet, Text, View, TouchableOpacity, Image} from 'react-native';
import Estilos from "../constants/Styles";
import Colors from "../constants/Colors";
import SociablyButton from "../components/SociablyButton";

const interests = ["Memes","Comedia","Entretenimiento","Deportes","Videojuegos","Bares","Teatro","Karaoke","Juegos de Mesa","Baile","Estudio","Conciertos"]

export default function InterestsScreen({navigation}) {

    const [tags,setTags] = useState([])

    function InterestButton(props){
        const texto = props.text
        const [toggle, setToggle] = useState(false)
        const toggleFunction = () => {
            setToggle(!toggle)
            let temp = tags
            temp.push(texto)
            setTags(temp)
        }
        return (
            <TouchableOpacity style={{...styles.interest,...toggle ? {borderColor: Colors.palette.primary}:{}}}>
                <Text style={{...styles.interestText,...toggle ? {color:Colors.palette.primary}:{}}} onPress={()=>{toggleFunction()}}>
                    {texto}
                </Text>
            </TouchableOpacity>
        )
    }

    function getTags(){
        let temp = tags
        let meme
        let res = []
        while (temp.length > 0){
            meme = temp.pop()
            if (temp.indexOf(meme) >= 0){
                temp.splice(temp.indexOf(meme),1)
            } else {
                res.push(meme)
            }
        }
        console.log(res);
        return res
    }


    return (
        <View style={styles.container}>
            <Image style={{width:"10rem", height:"8rem"}} source={require("../assets/icon.png")}/>
            <Text style={Estilos.Title}> Choose your interests </Text>
            <Text style={{...Estilos.Subtitle, marginBottom: 40}}> Personalize your feed </Text>
            <View style={styles.flex}>
                {interests.map((x) => (
                    <InterestButton text={x}/>
                ))}
            </View>
            <View style={{alignItems:'center',justifyContent:'center',width:'100%'}}>
                <SociablyButton buttonType={'lightButton'} text={"Skip"} onPress={()=>{console.log("Skiping")}}/>
                <SociablyButton buttonType={'darkButton'} text={"Next"} onPress={()=>{getTags()}}/>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },

    interest: {
        backgroundColor: '#fff',
        borderWidth: 1.5,
        borderColor: Colors.palette.secondary,
        borderRadius: 10,
        alignItems: 'center',
        justifyContent: 'center'
    },
    interestText:{
        fontSize: 21,
        fontFamily: "Space Grotesk",
        color: Colors.palette.secondary,
        paddingTop: 3,
        paddingBottom: 3,
        paddingLeft: 5,
        paddingRight: 5,
    },
    flex:{
        display: "flex",
        flexDirection: "row",
        flexWrap: "wrap",
        justifyContent: "center",
        alignItems: "flex-start",
        alignContent: "flex-start",
        gap: "0.9rem",
        height: "50vh"
    }
})
