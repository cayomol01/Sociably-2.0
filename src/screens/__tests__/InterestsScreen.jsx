import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, fireEvent} from '@testing-library/react-native'

import InterestsScreen from '../InterestsScreen';

test('renders correctly', () => {
    const tree = renderer.create(<InterestsScreen/>).toJSON();
    expect(tree).toMatchSnapshot();
});
