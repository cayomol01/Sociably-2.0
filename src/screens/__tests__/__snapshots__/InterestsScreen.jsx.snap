// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": "8rem",
        "width": "10rem",
      }
    }
  />
  <Text
    style={
      {
        "fontFamily": "Space Grotesk",
        "fontSize": "2rem",
        "fontWeight": "Bold",
      }
    }
  >
     Choose your interests 
  </Text>
  <Text
    style={
      {
        "fontFamily": "Space Grotesk",
        "fontSize": "1rem",
        "fontWeight": "Normal",
        "marginBottom": 40,
      }
    }
  >
     Personalize your feed 
  </Text>
  <View
    style={
      {
        "alignContent": "flex-start",
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "gap": "0.9rem",
        "height": "50vh",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Memes
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Comedia
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Entretenimiento
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Deportes
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Videojuegos
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Bares
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Teatro
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Karaoke
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Juegos de Mesa
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Baile
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Estudio
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#FF826C",
          "borderRadius": 10,
          "borderWidth": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FF826C",
            "fontFamily": "Space Grotesk",
            "fontSize": 21,
            "paddingBottom": 3,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 3,
          }
        }
      >
        Conciertos
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#C15548",
          "borderWidth": 0,
          "height": "6vh",
          "justifyContent": "center",
          "margin": "0.4rem",
          "opacity": 1,
          "text": {
            "color": "white",
            "fontFamily": "Space Grotesk",
            "fontSize": 26,
          },
          "width": "90%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Space Grotesk",
            "fontSize": 26,
          }
        }
      >
        Skip
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#C15548",
          "borderWidth": 2.7,
          "height": "6vh",
          "justifyContent": "center",
          "margin": "0.4rem",
          "opacity": 1,
          "text": {
            "color": "#C15548",
            "fontFamily": "Space Grotesk",
            "fontSize": 26,
          },
          "width": "90%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#C15548",
            "fontFamily": "Space Grotesk",
            "fontSize": 26,
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
